android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageView.kt

```diff
@@ -1,22 +1,612 @@
+						contentColor = Color.Black
+					),
+					onClick = { /*TODO*/ }) {
+
+					Text("View Details", style = MaterialTheme.typography.bodyLarge)
+
+				}
+				ButtonWithIcon(
+					modifier = Modifier
+						.fillMaxHeight(1.0f)
+						.width(dimensionResource(id = R.dimen.xxxl)),
+					colors = ButtonDefaults.buttonColors(
+						containerColor = darkerGrayButtonColor,
+						contentColor = Color.Black
+					),
+					iconId = R.drawable.chat_bubble_outline_gray_24,
+					onClick = { /*TODO*/ },
+					contentDescription = "chat icon"
+				)
+				ButtonWithIcon(
+					modifier = Modifier
+						.fillMaxHeight(1.0f)
+						.width(dimensionResource(id = R.dimen.xxxl)),
+					colors = ButtonDefaults.buttonColors(
+						containerColor = darkerGrayButtonColor,
+						contentColor = Color.Black
+					),
+					iconId = R.drawable.star_outline_black_26,
+					onClick = { /*TODO*/ },
+					contentDescription = stringResource(id = R.string.chat_icon)
+				)
+			}
+
+
+		}
+
+	}
+}
+
+@Composable
+fun FilterDropDown(
+	modifier: Modifier = Modifier,
+	filterName: String,
+	dropDownItems: Array<Int>,
+	updateState: (Int) -> Unit,
+	selectedValue: Int,
+	width: Dp,
 ) {
-	Column(
-		modifier = modifier.fillMaxSize(1.0f),
-		horizontalAlignment = Alignment.CenterHorizontally,
-		verticalArrangement = Arrangement.Center,
+	var expanded by remember { mutableStateOf(false) }
+
+	Box(
+		modifier = modifier
+			.width(width)
+			.height(dimensionResource(id = R.dimen.l))
+			.fillMaxWidth(1.0f)
 	) {
-		Text(text = "Home Page")
+		Button(
+			modifier = Modifier
+				.fillMaxWidth(1.0f)
+				.fillMaxHeight(1.0f),
+			contentPadding = PaddingValues(dimensionResource(id = R.dimen.s), ZERO_DP),
+			colors = ButtonDefaults.buttonColors(
+				containerColor = secondaryButtonBackgroundColor,
+				contentColor = secondaryTextColor,
+			),
+			shape = RoundedCornerShape(
+				dimensionResource(id = R.dimen.s)
+			),
+			onClick = {
+				expanded = !expanded
+			},
+		) {
+			Row(
+				modifier = Modifier.fillMaxWidth(1.0f),
+				verticalAlignment = Alignment.CenterVertically,
+				horizontalArrangement = Arrangement.SpaceBetween,
+			) {
+				Text(
+					text = filterName,
+					style = filterTextFont
+				)
+
+				Icon(
+					painter = painterResource(
+						id =
+						if (expanded) R.drawable.arrow_drop_up_soild_gray_24
+						else R.drawable.arrow_drop_down_solid_gray_24
+					),
+					contentDescription = stringResource(
+						id =
+						if (expanded) R.string.drop_up_arrow
+						else R.string.drop_down_arrow
+					)
+				)
+			}
+		}
+		DropdownMenu(
+			modifier = modifier
+
+				.background(Color.White),
+			expanded = expanded,
+			onDismissRequest = { expanded = false },
+		) {
+			dropDownItems.forEach {
+				DropdownMenuItem(
+					modifier = Modifier,
+					text = {
+						Text(
+							text = stringResource(it),
+							style = filterTextFont,
+							color = if (it == selectedValue && stringResource(it) != "Clear") subletrPink else Color.Black
```

Comment:
Maybe just compare the stringRes instead of `stringResource(it) != "Clear"` which will break in different locales.

Also, if we don't have one already, we should probably add a `darkTheme` adjusted primary text color.